### ISSUE TITLE
[rllib] PPO does not converge for CartPole with default config

### ISSUE THREAD

**Issue by ljendele** [ray-project/ray#7245]
*This is not a contribution.*

## What is the problem?
I noticed that Pytorch implementation does not manage to solve CartPole environment with the default config (the reward is stuck around 145). Tensorflow solves it in about 13 iterations both in Graph (no extra flag) and Eager mode (`--eager`)

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray version: newest from https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp36-cp36m-manylinux1_x86_64.whl
Python version: 3.6.10
Tensorflow version: 2.1
Pytorch version: 1.4.0
OS: Ubuntu 18.04 (ran in conda environment)

### Reproduction 
To reproduce the issue, simply run the following commands and observe the difference:
Pytorch - does not converge 
`rllib train --run=PPO --env=CartPole-v0 --torch`

Tensorflow - all of the following converge:
`rllib train --run=PPO --env=CartPole-v0 --config '{"simple_optimizer": true}' --eager`
`rllib train --run=PPO --env=CartPole-v0 --config '{"simple_optimizer": true}’`
`rllib train --run=PPO --env=CartPole-v0’`

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html).

Thanks, 

Lukas

**Comment by sven1977** [comment]
This is a strange initialization problem. PPO torch is performing at least as good as tf on Atari problems in terms of learning progress per timesteps (see https://github.com/ray-project/rl-experiments/tree/master/atari-ppo).
Yes, this test case sometimes flakes, but eventually does pass if you rerun it.
I'll see whether we can fine-tune this example differently.

**Comment by sven1977** [comment]
This is simply a tuning problem. E.g., try this yaml config and it'll work consistently within 30k timesteps in torch:

```
cartpole-ppo-torch:
    env: CartPole-v0
    run: PPO
    stop:
        episode_reward_mean: 150
        timesteps_total: 100000
    config:
        use_pytorch: true
        gamma: 0.99
        lr: 0.0003
        num_workers: 1
        observation_filter: MeanStdFilter
        num_sgd_iter: 6
        vf_share_layers: true
        vf_loss_coeff: 0.5
        model:
          fcnet_hiddens: [32]
          fcnet_activation: linear
```

I will update our tuned_example/regression_tests files for both torch and tf (learning is faster with the simplified configs above) and close this issue.

**Comment by sven1977** [comment]
This is the PR: https://github.com/ray-project/ray/pull/7556
